src/BinSkim.Rules/PERules/BA2022.SignSecurely.cs

```diff
@@ -157,17 +157,29 @@ public override void Analyze(BinaryAnalyzerContext context)
                         return false;
                     }
 
-                    default:
+                    case CryptoError.TRUST_E_CERT_SIGNATURE:
                     {
-                        string cryptoErrorDescription = cryptoError.GetErrorDescription();
-                        // '{0}' signing was flagged as insecure by WinTrustVerify with error code: '{1}' ({2})
-                        context.Logger.Log(this, RuleUtilities.BuildResult(FailureLevel.Error, context, null,
-                            nameof(RuleResources.BA2022_Error_DidNotVerify),
-                            context.CurrentTarget.Uri.GetFileName(),
-                            cryptoError.ToString(),
-                            cryptoErrorDescription));
+                        winTrustData = this.InitializeWinTrustDataStruct(filePath, WinTrustDataKind.EnforcePolicy, pszOID: Native.szOID_CERT_STRONG_KEY_OS_1);
```

Comment:
@shaopeng-gh If I understand correctly, it's the `Native.szOID_CERT_STRONG_KEY_OS_1` which makes the difference?

Reply:
correct, the definition is here
[https://msdn.microsoft.com/en-us/library/windows/desktop/hh870262(v=vs.85).aspx](https://msdn.microsoft.com/en-us/library/windows/desktop/hh870262(v=vs.85).aspx)
(this link is already in the solution \src\BinSkim.Rules\NativeInterop.cs)
It does not provide much details how WinVerifyTrust
deals with it. I can not find any public documents about it.
The source code of WinVerifyTrust is not public as well.